pkgs/tools/security/fprintd/default.nix

```diff
@@ -1,42 +1,70 @@
-{ thinkpad ? false
-, stdenv
-, fetchurl
+{ stdenv
+, fetchFromGitLab
 , fetchpatch
 , pkgconfig
-, intltool
-, libfprint-thinkpad ? null
-, libfprint ? null
+, meson
+, ninja
+, perl
+, gettext
+, libpam-wrapper
+, cairo
+, gtk-doc
+, libxslt
+, docbook-xsl-nons
+, docbook_xml_dtd_412
 , glib
 , dbus-glib
 , polkit
 , nss
 , pam
 , systemd
-, autoreconfHook
-, gtk-doc
+, libfprint
+, python37Packages
 }:
 
 stdenv.mkDerivation rec {
-  pname = "fprintd" + stdenv.lib.optionalString thinkpad "-thinkpad";
-  version = "0.9.0";
+  pname = "fprintd";
+  version = "1.90.1";
 
-  src = fetchurl {
-    url = "https://gitlab.freedesktop.org/libfprint/fprintd/uploads/9dec4b63d1f00e637070be1477ce63c0/fprintd-${version}.tar.xz";
-    sha256 = "182gcnwb6zjwmk0dn562rjmpbk7ac7dhipbfdhfic2sn1jzis49p";
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "libfprint";
+    repo = pname;
+    rev = version;
+    sha256 = "0mbzk263x7f58i9cxhs44mrngs7zw5wkm62j5r6xlcidhmfn03cg";
   };
 
+  outputs = [ "out" "devdoc" ];
+
   patches = [
+    # Fixes issue with ":" used when multiple paths (might be the case on NixOS)
+    # https://gitlab.freedesktop.org/libfprint/fprintd/-/merge_requests/50
     (fetchpatch {
-      url = "https://gitlab.freedesktop.org/libfprint/fprintd/merge_requests/16.patch";
-      sha256 = "1y39zsmxjll9hip8464qwhq5qg06c13pnafyafgxdph75lvhdll7";
+      url = "https://gitlab.freedesktop.org/libfprint/fprintd/-/merge_requests/50.patch";
+      sha256 = "1wqkk7bqig0fhjbq1487kxkh7yc9356qjsac1x27pzgakbayxpyy";
     })
   ];
 
+  pythonPath = with python37Packages; [
+    python-dbusmock
+    dbus-python
+    pygobject3
+    pycairo
+    libpam-wrapper
```

Comment:
This will super-likely lead to a python mismatch. Add it to as `pypamtest` to `python-packages.nix` and override the python there, see for example:

https://github.com/NixOS/nixpkgs/blob/28c2c0156da98dbe553490f904da92ed436df134/pkgs/top-level/python-packages.nix#L3059-L3075

Reply:
Noted, should I simply do an override like:

```nix
  pypamtest = pkgs.libpam-wrapper.override {
    enablePython = true;
    inherit python;
  };
```

Or should I split the output with a `py` one and add the whole `pipe` like `libselinux` ? I may have to patch/substitute some CMake files on `libpam-wrapper` because they always prefix the output path for python with the `CMAKE_INSTALL_PREFIX`.

Reply:
If this is only used for build/tests, splitting might be nice but not be necessary.

Reply:
`libpam-wrapper` is made for testing only. I will try to split it later if that's not a blocker :)